src/order/circular.lean

```diff
@@ -0,0 +1,387 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.basic
+
+/-!
+# Circular order hierarchy
+
+This file defines circular preorders, circular partial orders and circular orders.
+
+## Hierarchy
+
+* A `circular_order` is a ternary betweenness relation `btw : α → α → α → Prop`, which is
```

Comment:
```suggestion
* A `circular_order` is a ternary betweenness relation `btw : α → α → α → Prop` such that
  `btw a b c` states that `b` is between `a` and `c`.
```

Reply:
This is not *so* clear in my opinion because we must state what "between" means. When you take two points on a circle, aren't all the other ones between them? This depends on the orientation, and your proposed change fails to address that.

Reply:
If you're going to say that the english for `btw` is `between`, then you need to tell the reader which order to read the arguments too. My initial reading was that `btw a b c` means that `a` is between `b` and `c`.

Reply:
I expanded the subsequent explanation instead.

Reply:
Does it make sense to start this sentence as
```suggestion
* A ternary "betweenness" relation `btw : α → α → α → Prop` is a  *circular order* if it is:
```

Reply:
What about now?